apis/python/src/tiledbsoma/options/_soma_tiledb_context.py

```diff
@@ -31,15 +42,18 @@ class SOMATileDBContext:
 
     tiledb_ctx: tiledb.Ctx = _build_default_tiledb_ctx()
 
-    timestamp: Optional[int] = attrs.field(default=None)
+    timestamp_ms: Optional[int] = attrs.field(
+        default=None, converter=_maybe_timestamp_ms, alias="timestamp"
+    )
     """
-    Timestamp for operations on SOMA objects, in milliseconds since Unix epoch.
+    Default timestamp for operations on SOMA objects, in millis since the Unix epoch.
 
-    ``None``, the default, does not provide cross-object timestamp consistency:
+    This is used when a timestamp is not provided to an ``open`` operation.
 
-    - **Reads** will see data as of the time that the individual TileDB object
-      is opened.
-    - **Writes** will be recorded as of when the TileDB object is closed.
+    ``None``, the default, sets the timestamp on each root ``open`` operation.
+    That is, if you ``open`` a collection, and access individual members of the
+    collection through indexing or ``add_new``, the timestamp of all of those
+    operations will be that of the time you called ``open``.
```

Comment:
Per your response, "in cases where you want to use the same context many times, storing the timestamp there is bad because it means multiple open/write/close cycles will have no defined order.", do we want to warn against setting explicit timestamps in `SOMATileDBContext`? Or is that too edge-casey?

Reply:
This is a great point. I’ve added a warning here, recommending that users not do this. I would personally be OK with removing the timestamp from the context entirely, but didn’t want to go that far on this change.